s3 gateway API docs updates (#4431)
pachyderm/pachyderm: doc/docs/master/reference/s3gateway_api.md

trying to create the same bucket twice will return a `BucketAlreadyOwnedByYou`
error.

#### `DeleteObjects`

Route: `POST /<branch>.<repo>/?delete`.

Deletes multiple files specified in the request payload.

#### `DeleteObject`

Route: `DELETE /<branch>.<repo>/<filepath>`.

Deletes the PFS file `filepath` in an atomic commit on the HEAD of `branch`.

#### `GetObject`

Route: `GET /<branch>.<repo>/<filepath>`.

> By default, this gets the latest version of the given file (i.e. the file at

Reviewer: ```suggestion
By default, this request gets the latest version of the given file. For example, the file at the
```

> `HEAD` of the given repo/branch.) You can use s3's versioning API (i.e. with

Reviewer: ```suggestion
`HEAD` of the given repository and branch.) You can use the S3 versioning API, for example, 
```

> the `versionId` subresource) to get the object at a non-HEAD commit by

Reviewer: ```suggestion
the `versionId` subresource, to get the object at a non-HEAD commit by
```

> specifying either a specific commit ID, or via the caret syntax (e.g.

Reviewer: ```suggestion
specifying either a specific commit ID, or by using the caret syntax. For example,
```

> `HEAD^`.)

Reviewer: ```suggestion
`HEAD^`.
```
